src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs

```diff
@@ -0,0 +1,77 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Cdn.Common;
+using Microsoft.Azure.Commands.Cdn.Helpers;
+using Microsoft.Azure.Commands.Cdn.Models.CustomDomain;
+using Microsoft.Azure.Commands.Cdn.Properties;
+using Microsoft.Azure.Management.Cdn;
+using System.Linq;
+using Microsoft.Azure.Commands.Cdn.Models.Endpoint;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Cdn.CustomDomain
+{
+    [Cmdlet(VerbsLifecycle.Disable, "AzureRmCdnCustomDomain", DefaultParameterSetName = FieldsParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
```

Comment:
@prakharsharma10 this cmdlet says it implements `ShouldProcess`, but there is no `ShouldProcess` method call wrapping the cmdlet execution below, so if a user provides `-WhatIf` to see what will happen when they run this command, it will actually run the command, which isn't the experience we want. Check out [this document](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/should-process-confirm-impact.md) for more information on using this method.

Reply:
@cormacpayne Added ConfirmAction method as done for other cmdlets in CDN that supports shouldProcess